indexer/ftypes_sponsored.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "indexer/ftypes_matcher.hpp"
+
+#include <string>
+
+namespace ftypes
+{
+class BaseSponsoredChecker : public BaseChecker
+{
+protected:
+  explicit BaseSponsoredChecker(std::string const & sponsoredType);
+};
+
+class IsBookingChecker : public BaseSponsoredChecker
+{
+  IsBookingChecker() : BaseSponsoredChecker("booking") {}
+public:
+  DECLARE_CHECKER_INSTANCE(IsBookingChecker);
+};
+
+class IsOpentableChecker : public BaseSponsoredChecker
```

Comment:
Looks like you may also wrap declarations of these small checkers into macro :)

Reply:
+1